pages/level/[[...params]].tsx

```diff
@@ -87,23 +97,22 @@ interface LevelSWRProps {
 
 export default function LevelSWR({ level }: LevelSWRProps) {
   const router = useRouter();
-  const { id } = router.query;
-
-  if (router.isFallback || !id) {
+  if (router.isFallback || !level) {
     return <SkeletonPage/>;
   }
-
+  const id = level._id;
+  const levelProps = {level:level};
   return (
     <SWRConfig value={{ fallback: { [getSWRKey(`/api/level-by-id/${id}`)]: level } }}>
-      <LevelPage/>
+      <LevelPage {...levelProps} />
     </SWRConfig>
   );
 }
 
-function LevelPage() {
+function LevelPage(levelProps:any) {
   const router = useRouter();
-  const { id, wid } = router.query;
-  const { level } = useLevelById(id);
+  const { wid } = router.query;
+  const { level } = levelProps;
```

Comment:
Should use useLevelsByUsernameAndSlug here. Although depending on if both the id and slug url formats are supported, we should maybe change it to a generic useLevel() that calls the correct API depending on params?

Reply:
What do these use functions do exactly? It seems to work without it but am I missing a flow that needs to use it?

Reply:
`SWRConfig value={{ fallback: { [getSWRKey(`/api/level-by-id/${id}`)]: level } }}`

this code is passing in the "stale" value for the API (the statically generated level), and then `useLevelById(id)` is actually running the API call to update the value of `level`